website/pages/docs/agent/config-entries/service-intentions.mdx

```diff
@@ -0,0 +1,310 @@
+  value](/docs/connect/intentions#precedence-and-match-order) computed from the
+  source and destination naming components.
+
+- `Type` `(string: <required>)` - The type for the `Name` value. This can be
+  only `"consul"` today to represent a Consul service.
+
+- `Description` `(string: "")` - Description for the intention. This is not
+  used for anything by Consul, but is presented in API responses to assist
+  tooling.
+
+- `LegacyID` `(string: <read-only>)` - This is the UUID to uniquely identify
+  this intention in the system. Cannot be set directly and is exposed here as an artifact of
+  the config entry migration and is primarily used to allow legacy intention 
+  [API](/api-docs/connect/intentions#update-intention-by-id)
+  [endpoints](/api-docs/connect/intentions#read-specific-intention-by-id)
+  to continue to function for a period of time after ugprading to 1.9.0.
+  TODO(link-do-long-form-docs)
+
+- `LegacyMeta` `(map<string|string>: <read-only>)` - Specified arbitrary KV
+  metadata pairs attached to the intention, rather than to the enclosing config
+  entry. Cannot be set directly and is exposed here as an artifact of the
+  config entry migration and is primarily used to allow legacy intention
+  [API](/api-docs/connect/intentions#update-intention-by-id)
+  [endpoints](/api-docs/connect/intentions#read-specific-intention-by-id) to
+  continue to function for a period of time after ugprading to 1.9.0.
+  TODO(link-do-long-form-docs)
+
+- `LegacyCreateTime` `(time: optional)` - The timestamp that this intention was
+  created. Cannot be set directly and is exposed here as an artifact of the
+  config entry migration and is primarily used to allow legacy intention
+  [API](/api-docs/connect/intentions#update-intention-by-id)
+  [endpoints](/api-docs/connect/intentions#read-specific-intention-by-id) to
+  continue to function for a period of time after ugprading to 1.9.0.
+  TODO(link-do-long-form-docs)
+
+- `LegacyUpdateTime` `(time: optional)` - The timestamp that this intention was
+  last updated. Cannot be set directly and is exposed here as an artifact of
+  the config entry migration and is primarily used to allow legacy intention
+  [API](/api-docs/connect/intentions#update-intention-by-id)
+  [endpoints](/api-docs/connect/intentions#read-specific-intention-by-id) to
+  continue to function for a period of time after ugprading to 1.9.0.
+  TODO(link-do-long-form-docs)
+
+### `IntentionPermission`
+
+- `Action` `(string: <required>)` - This is one of "allow" or "deny" for
+  the action that should be taken if this permission matches a request.
+
+- `HTTP` `(IntentionHTTPPermission: <required>)` - A set of
+  [http-specific authorization criteria](#intentionhttppermission).
+
+### `IntentionHTTPPermission`
+
+- `PathExact` `(string: "")` - Exact path to match on the HTTP request path.
+
+  At most only one of `PathExact`, `PathPrefix`, or `PathRegex` may be configured.
```

Comment:
Do we usually include input field requirements/restrictions here? i.e. accepted characters?

Reply:
Yes, and these have no restrictions that consul enforces currently other than "PathExact/PathPrefix must start with a leading slash", just like service-router. Practically speaking envoy is going to complain if these make no sense at that layer, but they're largely just opaque string fields to Consul.

Reply:
Oh that's an interesting requirement. We could add the slash for the user if they forget it (in a future iteration of this project) cc @jnwright

Reply:
John has already added an error message if users forget to add the slash 🎉